packages/gatsby/src/schema/infer/add-inferred-fields.js

```diff
@@ -212,7 +213,9 @@ const getFieldConfigFromFieldNameConvention = (
     const typeName = linkedTypes.sort().join(``) + `Union`
     type = schemaComposer.getOrCreateUTC(typeName, utc => {
       const types = linkedTypes.map(typeName =>
-        schemaComposer.getOrCreateTC(typeName)
+        schemaComposer.getOrCreateTC(typeName, tc => {
```

Comment:
Why? All node types should be created before inference.

Reply:
Mostly for testing (when we don't use the whole pipeline). I guess doesn't hurt to have it there

Reply:
Ok, so I think I've added that addNodeInterface thing as a hack so I could distinguish between typeDefs that erroneously don't have implements. I guess it should actually be in https://github.com/gatsbyjs/gatsby/blob/d45c97c31e198aceb5c9f39928e907339a59fa8c/packages/gatsby/src/schema/infer/index.js

Reply:
I'll make changes to fix that. Also I fill add "createGatsbySchemaComposer" function that adds default types and directives, that we can use in tests too.